concepts/observability-configurable-logging/07-multiple-logpipeline-investigation.md

```diff
@@ -0,0 +1,106 @@
+1. Installed kyma with telemetry operator
+2. Two outputs were deployed in the kyma cluster
+    - Loki which comes with kyma
+    - Mockserver [deployment](./assets/logpipeline-invstigation/mock-server.yaml)
+3. Log generator [daemon set](./assets/logpipeline-invstigation/log-generator.yaml) to generate huge amount of logs to saturate the buffer faster.
+4. [Function](./assets/logpipeline-invstigation/func.js) to check if the logs are being delivered when one of the output is down.
+5. To simulate failures the port of the service was changed so that dns resolution would still work but logs wont be deliverd.
+
+## Test Cases
+
+### Case 1
+![a](./assets/logpipeline-invstigation/case-1/case-1.svg)
+
+Setup:
+1. one input and 2 outputs (with limiting max number of chunks in filesystem) without rewrite tags.
+2. logpipelines: [loki](./assets/logpipeline-invstigation/case-1/loki.yaml), [mock-server](./assets/logpipeline-invstigation/case-1/mockserver.yml)
+
+Result:
+1. When one of the output is down then we see the filesystem buffer is being filled up and would only keep 150M of latest data (old data would purged). Eventually the output plugin for loki stops as well as the buffer at tail plugin is full.
+2. When both down then tail plugin stopped is stopped after filling up the buffer. Although the buffer was not fully filled (104m/150M). Still the tail plugin was stopped
+
+### Case 2
+![a](./assets/logpipeline-invstigation/case-2/case-2.svg)
+
+Setup:
+1. one input and 2 outputs (loki (fluentbit-loki plugin) + mockserver) with rewrite-tags
+2. logpipelines: [loki](./assets/logpipeline-invstigation/case-2/loki.yaml), [mock-server](./assets/logpipeline-invstigation/case-2/mockserver.yml)
+
+
+Result:
+1. Output chunking keeps with 150M of newest data.
+2.  The tail plugin sends the data to next phase of pipeline (filter plugin). If the outputs is not working then only the buffer at the rewrite tag would be filled and tail plugin buffer wont be filled
+3. When the buffer is full then we see some error logs stating that and the old logs are discarded
+    ```unix
+    [2022/04/21 14:38:23] [error] [input:emitter:log_emitter] error registering chunk with tag: log_rewritten
+    [2022/04/21 14:38:24] [error] [input chunk] chunk 1-1650551903.999375896.flb would exceed total limit size in plugin http.1
+    ```
+4. When grafana-loki plugin is stopped it stopped all the pipelines (so mockserver which was fine was also stopped) 
+5. When mock server was down the logs were still shipped to loki
+
+
+### Case 3
+
+Setup:
+1. one input with 2 outputs (mockserver + mockserver) with rewrite tags
+2. logpipelines: [mockserver-1](./assets/logpipeline-invstigation/case-3/mockserver-1.yml), [mockserver-2](./assets/logpipeline-invstigation/case-3/mockserver-2.yml)
+
+
+Result
+1. When both outputs are down then the buffer was filled and eventually the fluentbit pod got killed because of 500 (most probably because of CPU throttling)
```

Comment:
If the order doesn't matter, use an unordered list

```suggestion
- When both outputs are down, the buffer was filled and eventually the Fluent Bit Pod went down with error code `500` (most probably because of CPU throttling).
```